src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -856,7 +858,7 @@ class HookUtils(object):
             pbs.logmsg(pbs.EVENT_DEBUG4,
                        'assigned_resources: %s' %
                        (cgroup.assigned_resources))
-            cgroup.setup_job_devices_env()
+            cgroup.setup_job_devices_env(node.devices['gpu'])
```

Comment:
Would it be possible that node.devices does not have a key and value for 'gpu'? If so, it would cause an exception here. Maybe we need to check if key exists first?

Reply:
`node.devices` is populated by `_discover_devies()` when a NodeConfig instance is created. And `_discover_devies()` always initialize `self.devices['gpu']` to be an empty dictionary.  So this should be safe since the key `gpu` always exist. 

See line 1430, in function `_discover_devices`.

Reply:
Looks good.